compiler/rustc_ast_pretty/src/pprust/state.rs

```diff
@@ -240,9 +241,37 @@ pub fn print_crate<'a>(
     edition: Edition,
     g: &AttrIdGenerator,
 ) -> String {
-    let mut s =
-        State { s: pp::Printer::new(), comments: Some(Comments::new(sm, filename, input)), ann };
+    let mut s = State {
+        s: pp::Printer::new(),
+        comments: Some(Comments::new(sm, filename, input)),
+        ann,
+        is_interface: false,
+    };
+
+    print_crate_inner(&mut s, krate, is_expanded, edition, g);
+
+    s.ann.post(&mut s, AnnNode::Crate(krate));
+    s.s.eof()
+}
+
+pub fn print_crate_as_interface(
```

Comment:
It's ok to add a new flag to `fn print_crate` instead of introducing a new function.

Reply:
I would prefer a new function. For the interface dump, `PpAnn` must always be fixed and comments must always be skipped.

Reply:
Ok.